ocw_data_parser/ocw_data_parser_test.py

```diff
@@ -2,29 +2,189 @@
+        yield s3_bucket
+
+@pytest.fixture(autouse=True, scope="function")
+def ocw_parser():
+    """
+    Instantiate an OCWParser object and run functions depending on args passed in
+    """
+    with TemporaryDirectory() as destination_dir:
+        parser = OCWParser(course_dir=constants.COURSE_DIR,
+                            destination_dir=destination_dir,
+                            static_prefix=constants.STATIC_PREFIX)
+        yield parser
+
+@pytest.fixture(autouse=True, scope="session")
+def ocw_parser_s3():
+    with TemporaryDirectory() as destination_dir:
+        parser = OCWParser(course_dir=constants.COURSE_DIR,
+                            destination_dir=destination_dir,
+                            static_prefix=constants.STATIC_PREFIX)
+        parser.setup_s3_uploading(
+            s3_bucket_name="testing",
+            s3_bucket_access_key="testing",
+            s3_bucket_secret_access_key="testing",
+            folder="testing"
+        )
+        yield parser
+
+@pytest.fixture(autouse=True)
+def course_id(ocw_parser):
+    yield ocw_parser.master_json["short_url"]
+
+def test_no_params(ocw_parser):
+    """
+    Test that an OCWParser with no params raises an exception
+    """
+    with pytest.raises(Exception):
+        OCWParser()
+
+def test_html_parser_output_list(ocw_parser):
+    """
+    Test passing in an output_list to the CustomHTMLParser
+    """
+    output_list = ["test"]
+    parser = CustomHTMLParser(output_list=output_list)
+    assert "test" == parser.output_list[0]
+
+def test_parser_loaded_jsons(ocw_parser):
+    """
+    Test instantiating a parser with preloaded jsons
+    """
+    try:
+        assert OCWParser(loaded_jsons=ocw_parser.jsons)
+    except:
+        fail("instantiating parser with preloaded jsons failed")
+
+def test_generate_master_json_none_source(ocw_parser):
+    """
+    Make sure that running generate_master_json doesn't throw an error after nulling 
+    out the parser's source jsons
+    """
+    ocw_parser.jsons = None
+    ocw_parser.generate_master_json()
+    assert ocw_parser.jsons is not None
+
+def test_generate_master_json_none_course_image_uid(ocw_parser):
+    """
+    Make sure course_image_uid is regenerated by generate_master_json after it's nulled out
+    """
+    ocw_parser.course_image_uid = None
+    ocw_parser.generate_master_json()
+    assert ocw_parser.course_image_uid is not None
+
+def test_load_raw_jsons_invalid_file(ocw_parser):
+    """
+    Add a json file with invalid content to the course_dir and make sure it generates an error
+    """
+    with TemporaryDirectory() as destination_dir:
+        with open(os.path.join(constants.COURSE_DIR, "jsons/999.json"), "w") as f:
+            f.write("{")
+        try:
+            OCWParser(course_dir=constants.COURSE_DIR,
+                    destination_dir=destination_dir,
+                    static_prefix=constants.STATIC_PREFIX)
+            assert False
```

Comment:
To assert an exception you can use `pytest.raises`:

    with pytest.raises(Exception):
        OCWParser(course_dir=constants.COURSE_DIR,
            destination_dir=destination_dir,
            static_prefix=constants.STATIC_PREFIX)